src/Extension/Attribute/AsTwigExtension.php

```diff
@@ -0,0 +1,20 @@
+<?php
+
+/*
+ * This file is part of Twig.
+ *
+ * (c) Fabien Potencier
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Twig\Extension\Attribute;
+
+/**
+ * Identifies a class that uses PHP attributes to define filters, functions, or tests.
+ */
+#[\Attribute(\Attribute::TARGET_CLASS)]
+class AsTwigExtension
```

Comment:
What is the purpose of this attribute ? I don't see (yet) how it is / will be used

Reply:
Given that extensions must implement the ExtensionInterface anyway, there is no need for a marker attribute. The Symfony dependency injection component is also able to perform auto-configuration based on implemented interfaces (and TwigBundle already does it for the Twig ExtensionInterface

Reply:
It's more a question of DX. It's more meaningful to use an attribute than an interface as a marker. Implementing an interface without method is a deviation from the interface system.

Reply:
I've made this attribute mandatory to prevent any object from being passed by mistake. In practice, it will be mandatory anyway for Symfony autoconfiguration.